### PR TITLE
fix hangfire dynamic proxy serialization error

### PR DESCRIPTION
**Pull request by Nokecy**
1、https://github.com/HangfireIO/Hangfire/issues/558#issuecomment-821957356
2、add AutoMapper.Collection

### CODE REVIEW

File: framework/src/Volo.Abp.BackgroundWorkers.Hangfire/Volo/Abp/BackgroundWorkers/Hangfire/HangfireBackgroundWorkerManager.cs [hunk 1]
```diff
@@ -73,6 +74,21 @@ public Task AddAsync(IBackgroundWorker worker)
...
+            realType = ProxyUtil.GetUnproxiedType(instance);
+        }
+        else
+        {
+            realType = instance.GetType();
+        }
+
+        return realType;
+    }
+
```
Comment (maliming) [inline framework/src/Volo.Abp.BackgroundWorkers.Hangfire/Volo/Abp/BackgroundWorkers/Hangfire/HangfireBackgroundWorkerManager.cs:91]: https://github.com/abpframework/abp/blob/e3e1779de6df5d26f01cdc8e99ac9cbcb3d24d3c/framework/src/Volo.Abp.Core/Volo/Abp/DynamicProxy/ProxyHelper.cs#L34

File: framework/src/Volo.Abp.AutoMapper/Volo/Abp/AutoMapper/AbpAutoMapperModule.cs [hunk 2]
```diff
@@ -53,10 +54,12 @@ void ValidateAll(IConfigurationProvider config)
...
-            var mapperConfiguration = new MapperConfiguration(mapperConfigurationExpression =>
-            {
-                ConfigureAll(new AbpAutoMapperConfigurationContext(mapperConfigurationExpression, scope.ServiceProvider));
-            });
+                var mapperConfiguration = new MapperConfiguration(mapperConfigurationExpression =>
+                {
+                    mapperConfigurationExpression.AddCollectionMappers();
+
+                    ConfigureAll(new AbpAutoMapperConfigurationContext(mapperConfigurationExpression, scope.ServiceProvider));
+                });
```
Comment (maliming) [inline framework/src/Volo.Abp.AutoMapper/Volo/Abp/AutoMapper/AbpAutoMapperModule.cs:63]: Please do the AutoMapper changes in another PR.

**Review by maliming — CHANGES_REQUESTED** [review]

**Comment by maliming** [comment]
Can you open PR on the `rel-5.0` branch?

**Comment by realLiangshiwei** [comment]
We also need to change Quartz. I will handle it. thanks @Nokecy

**Comment by Nokecy** [comment]
ok ~~ @realLiangshiwei

**Comment by realLiangshiwei** [comment]
@Nokecy 

I don't think we need `Automapper.Collection`, you can configure it in your project:

```csharp
Configure<AbpAutoMapperOptions>(options =>
{
    options.Configurators.Add(x =>
    {
        x.MapperConfiguration.AddCollectionMappers();
    });
});
```
